go module example failing

hello! the main.go example in _example didn't work for me, dependencies may have drifted and there was no mod or lock. specifically the charmbracelet/keygen library seems to be different? and then log doesnt have ln( like println( anymore. but something with the keygen changes i tried to do caused pointer errors once it compiled and i ran it. i get a segfault with both wsl ubuntu and windows 11. i havent tried a regular linux box.

```
user@laptop-framework:/mnt/c/Users/drewr/code/sources/launcher$ ./launcher
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6b7e07]

goroutine 1 [running]:
github.com/charmbracelet/keygen.New({0xc0001482e0?, 0x0?}, {0xc00018be80, 0x2, 0x0?})
        /home/user/go/pkg/mod/github.com/charmbracelet/keygen@v0.5.0/keygen.go:167 +0x107
main.main()
        /mnt/c/Users/drewr/code/sources/launcher/main.go:20 +0xc5

  ❯ .\launcher.exe
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x1279927]

goroutine 1 [running]:
github.com/charmbracelet/keygen.New({0xc0001b4840?, 0x2?}, {0xc0001d1e80, 0x2, 0x0?})
        C:/Users/drewr/go/pkg/mod/github.com/charmbracelet/keygen@v0.5.0/keygen.go:167 +0x107
main.main()
        C:/Users/drewr/code/src/launcher/main.go:20 +0xc5
```

go 1.21.3



```
// ...
	k, err := keygen.New(filepath.Join(".wishlist", "server"), nil, keygen.WithKeyType(keygen.Ed25519)) //<-- this is different

// ...
			return wish.NewServer(
				wish.WithAddress(e.Address),
				wish.WithHostKeyPEM(k.RawProtectedPrivateKey()), // <-- this is different
				wish.WithPublicKeyAuth(func(ctx ssh.Context, key ssh.PublicKey) // ...
```

and then the log line changes

--- a/_example/main.go
+++ b/_example/main.go
@@ -3,10 +3,12 @@ package main
 import (
 	"fmt"
 	"path/filepath"
+	"time"
 
 	"github.com/charmbracelet/bubbles/spinner"
 	tea "github.com/charmbracelet/bubbletea"
 	"github.com/charmbracelet/keygen"
+	"github.com/charmbracelet/lipgloss"
 	"github.com/charmbracelet/log"
 	"github.com/charmbracelet/ssh"
 	"github.com/charmbracelet/wish"
@@ -17,22 +19,26 @@ import (
 )
 
 func main() {
-	k, err := keygen.New(filepath.Join(".wishlist", "server"), nil, keygen.Ed25519)
+	k, err := keygen.New(
+		filepath.Join(".wishlist", "server"),
+		keygen.WithKeyType(keygen.Ed25519),
+	)
 	if err != nil {
-		log.Fatalln(err)
+		log.Fatal("Server keypair", "err", err)
 	}
 	if !k.KeyPairExists() {
 		if err := k.WriteKeys(); err != nil {
-			log.Fatalln(err)
+			log.Fatal("Server keypair", "err", err)
 		}
 	}
 
 	// wishlist config
 	cfg := &wishlist.Config{
+		Port: 2233,
 		Factory: func(e wishlist.Endpoint) (*ssh.Server, error) {
 			return wish.NewServer(
 				wish.WithAddress(e.Address),
-				wish.WithHostKeyPEM(k.PrivateKeyPEM()),
+				wish.WithHostKeyPEM(k.RawPrivateKey()),
 				wish.WithPublicKeyAuth(func(ctx ssh.Context, key ssh.PublicKey) bool {
 					return true
 				}),
@@ -49,8 +55,9 @@ func main() {
 			{
 				Name: "bubbletea",
 				Middlewares: []wish.Middleware{
-					bm.Middleware(func(ssh.Session) (tea.Model, []tea.ProgramOption) {
-						return initialModel(), nil
+					bm.Middleware(func(s ssh.Session) (tea.Model, []tea.ProgramOption) {
+						r := bm.MakeRenderer(s)
+						return initialModel(r), nil
 					}),
 				},
 			},
@@ -60,6 +67,7 @@ func main() {
 					func(h ssh.Handler) ssh.Handler {
 						return func(s ssh.Session) {
 							_, _ = s.Write([]byte("hello, world\n\r"))
+							time.Sleep(time.Second * 5)
 							h(s)
 						}
 					},
@@ -89,34 +97,35 @@ func main() {
 
 	// start all the servers
 	if err := wishlist.Serve(cfg); err != nil {
-		log.Fatalln(err)
+		log.Fatal("Serve", "err", err)
 	}
 }
 
 type model struct {
-	spinner spinner.Model
+	spinner  spinner.Model
+	renderer *lipgloss.Renderer
 }
 
-func initialModel() model {
-	s := spinner.NewModel()
+func initialModel(r *lipgloss.Renderer) model {
+	s := spinner.New()
 	s.Spinner = spinner.Dot
-	return model{spinner: s}
+	return model{spinner: s, renderer: r}
 }
 
 func (m model) Init() tea.Cmd {
-	return spinner.Tick
+	return m.spinner.Tick
 }
 
 func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 	switch msg := msg.(type) {
 	case tea.KeyMsg:
-		log.Println("keypress:", msg)
+		log.Info("keypress", "msg", msg)
 		switch msg.String() {
 		case "q", "esc", "ctrl+c":
 			return m, tea.Quit
 		}
 	case tea.WindowSizeMsg:
-		log.Println("window size:", msg)
+		log.Info("window size", "msg", msg)
 	}
 	var cmd tea.Cmd
 	m.spinner, cmd = m.spinner.Update(msg)
@@ -124,6 +133,9 @@ func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 }
 
 func (m model) View() string {
-	str := fmt.Sprintf("\n\n   %s Loading forever...press q to quit\n\n", m.spinner.View())
+	str := fmt.Sprintf(
+		"\n\n   %s Loading forever... press q to quit\n\n",
+		m.renderer.NewStyle().Foreground(lipgloss.Color("5")).Render(m.spinner.View()),
+	)
 	return str
 }

